fiftyone/server/routes/get_similar_labels_frames.py

```diff
@@ -0,0 +1,312 @@
+"""
+FiftyOne Server /get_similar_labels_frames route
+
+| Copyright 2017-2025, Voxel51, Inc.
+| `voxel51.com <https://voxel51.com/>`_
+|
+"""
+
+from bson import ObjectId
+from starlette.endpoints import HTTPEndpoint
+from starlette.requests import Request
+from starlette.responses import JSONResponse
+
+import fiftyone.core.labels as fol
+import fiftyone.core.odm as foo
+import fiftyone.core.view as fov
+import fiftyone.server.view as fosv
+from fiftyone.core.expressions import ViewField as F
+from fiftyone.core.utils import run_sync_task
+from fiftyone.server.decorators import route
+
+
+class GetSimilarLabelsFrameCollection(HTTPEndpoint):
+    """
+    This route scans the frame collection for a given video sample and returns
+    a label id map (label_id -> frame_number) for a given instance id.
+    """
+
+    @route
+    async def post(self, request: Request, data: dict):
+        instance_id = data.get("instanceId")
+
+        if not instance_id:
+            return JSONResponse(
+                {
+                    "error": "instanceId is required",
+                },
+                status_code=400,
+            )
+
+        num_frames_str = data.get("numFrames")
+
+        if not num_frames_str:
+            return JSONResponse(
+                {
+                    "error": "numFrames is required",
+                },
+                status_code=400,
+            )
+
+        try:
+            num_frames = int(num_frames_str)
+
+            if num_frames <= 0:
+                raise ValueError
+        except ValueError:
+            return JSONResponse(
+                {
+                    "error": "numFrames must be a positive integer greater than 0",
+                },
+                status_code=400,
+            )
+
+        dataset = data.get("dataset")
+
+        if not dataset:
+            return JSONResponse(
+                {
+                    "error": "dataset is required",
+                },
+                status_code=400,
+            )
+
+        sample_id = data.get("sampleId")
+
+        if not sample_id:
+            return JSONResponse(
+                {
+                    "error": "sampleId is required",
+                },
+                status_code=400,
+            )
+
+        stages = data.get("view", [])
+        extended = data.get("extended", None)
+
+        start_frame = 1
+
+        end_frame = num_frames
+
+        support = None if stages else [start_frame, end_frame]
+
+        def get_frames_view(view):
+            view = fov.make_optimized_select_view(
+                view, sample_id, flatten=True
+            )
+
+            view = view.set_field(
+                "frames",
+                F("frames").filter(
+                    (F("frame_number") >= start_frame)
+                    & (F("frame_number") <= end_frame)
+                ),
+            )
```

Comment:
This `set_field` can be removed because `support` is used in the `_pipeline()` call

Reply:
thanks. fixed